### PR TITLE
[tools/depends] Bump Python 3.12.5

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
Bump tools/depends platforms python to 3.12.5

## Motivation and context
There are some large changes coming regarding cpython that will affect ios/tvos/android considerably in the way we currently build, and will require rather large changes.
I figure i need to get closer to 3.13 to minimise the grief

Major changes in 3.12
- distutils removal from stdlib. It now requires setuptools.
  A number of patches (samba/mesa) are done to handle this change. I dont want to fuck with mesa updates, and samba updates have proven extremely difficult for me to get going. 
- All sorts of build options have changed. eg modules.setup no longer "needed". It can be valid, but it is processed after modules.stdlib which we would have to patch things in/out. Was just easer to move to ``py_cv_module_*`` exclusions.

A major note is the inclusion of a patch that was denied upstream for the 3.12 branch. This is in relation to https://github.com/python/cpython/issues/116510
A backport of the 3.13+ fix https://github.com/python/cpython/pull/118618 was done by a python developer, but was rejected for merge as they felt it was too risky. We have already seen linux users using python 3.12 getting trouble because of this issue, so i think we are best to include the commits anyway to resolve it for android/apple platforms. There were reports from linux kodi users that the commits did fix crashes.

As is always the case with python version bumps, im sure some shit gonna be broken, so i figure the sooner this gets in, the sooner the cleanup for those inevitable issues can be found and made.

One other note. For bisection, Most of this would need to be a single commit (native/target python, mesa, samba commits). Im not sure how best to handle that. Do i squash it all?

## How has this been tested?
Jenkins build for all platforms
android aarch64 runtime used some video plugins and seemed to work

## What is the effect on users?
N/A

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by fuzzard** [comment]
Jenkins build this please

**Review by garbear — APPROVED** [review]

File: tools/depends/native/pythonmodule-setuptools/PYTHONMODULE-SETUPTOOLS-VERSION [hunk 1]
```diff
@@ -0,0 +1,4 @@
+LIBNAME=setuptools
+VERSION=72.1.0
```
Comment (garbear) [inline tools/depends/native/pythonmodule-setuptools/PYTHONMODULE-SETUPTOOLS-VERSION:2]: Version 72.2.0 was released two days ago

Comment (fuzzard) [reply tools/depends/native/pythonmodule-setuptools/PYTHONMODULE-SETUPTOOLS-VERSION:2]: Dont care, im not chasing the release dragon. its been setup/tested as is, thats what im going with

**Comment by emveepee** [comment]
Is the Windows version of python going to continue at 3.8.15 in Piers?  The further the python versions get apart the harder it is to make plugins compatible as deprecated functions get removed https://docs.python.org/3/whatsnew/3.12.html#removed.  Although I understand the system python might be used in some cases this difference eventually will need to be addressed.

**Comment by fuzzard** [comment]
Prs are always open to someone dealing with it

**Comment by emveepee** [comment]
> Prs are always open to someone dealing with it

Right I personally don't care, but bumping the non-Windows builds is the cause of the problem so it should be considered.

**Comment by graysky2** [comment]
> A backport of the 3.13+ fix https://github.com/python/cpython/pull/118618 was done by a python developer, but was rejected for merge as they felt it was too risky. We have already seen linux users using python 3.12 getting trouble because of this issue, so i think we are best to include the commits anyway to resolve it for android/apple platforms. There were reports from linux kodi users that the commits did fix crashes.

I can confirm that this does fix the kodi segfault on exit.  I tested vanilla python 3.12.5 and the segfaults are still present.  Patching 3.12.5 with that PR fixes it.

**Review by wsnipex — APPROVED** [review]

**Pull request merged**